org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientXtextIntegrationTest.xtend

```diff
@@ -0,0 +1,160 @@
+import javax.ws.rs.client.Entity
+import javax.ws.rs.client.Invocation.Builder
+import javax.ws.rs.client.WebTarget
+import javax.ws.rs.core.GenericType
+import org.eclipse.jetty.server.session.SessionHandler
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.resource.EObjectDescription
+import org.eclipse.xtext.scoping.IGlobalScopeProvider
+import org.eclipse.xtext.scoping.IScopeProvider
+import org.eclipse.xtext.testing.XtextRunner
+import org.eclipse.xtext.testing.util.ParseHelper
+import org.junit.ClassRule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.testeditor.aml.dsl.AmlRuntimeModule
+import org.testeditor.aml.dsl.AmlStandaloneSetup
+import org.testeditor.tcl.ComponentTestStepContext
+import org.testeditor.tcl.TclFactory
+import org.testeditor.tcl.TclModel
+import org.testeditor.tcl.TclPackage
+import org.testeditor.tcl.dsl.TclRuntimeModule
+import org.testeditor.tcl.dsl.TclStandaloneSetup
+import org.testeditor.tsl.dsl.TslRuntimeModule
+import org.testeditor.tsl.dsl.web.TslWebSetup
+import org.testeditor.web.backend.xtext.TestEditorApplication
+import org.testeditor.web.backend.xtext.TestEditorConfiguration
+import org.testeditor.web.dropwizard.xtext.XtextApplication
+import org.testeditor.web.dropwizard.xtext.XtextServiceResource
+
+import static org.assertj.core.api.Assertions.assertThat
+import static org.mockito.ArgumentMatchers.*
+import static org.mockito.Mockito.*
+
+@RunWith(XtextRunner)
+//@InjectWith(IntegrationTestInjectorProvider)
+class IndexServiceClientXtextIntegrationTest {
+
+	@ClassRule
+	public static val dropwizardClient = new DropwizardAppRule(TestEditorTestApplication,
+		ResourceHelpers.resourceFilePath("config.yml"), #[ // config('logging.level', 'TRACE')
+		])
+
+	val injector = (dropwizardClient.application as TestEditorTestApplication).getTclInjector
+
+	val extension ParseHelper<TclModel> parserHelper = injector.getInstance(
+		Key.get(new TypeLiteral<ParseHelper<TclModel>>() {
+		}))
+	val extension IScopeProvider scopeProvider = injector.getInstance(IScopeProvider)
+
+	@Test
+	def void shouldCompleteScopeRequest() {
+		// given
+		val context = '''
+			package pack
+			
+			# MacroLib
+			
+			## FirstMacro
+			
+				template = "code"
+			
+				Component: SomeComponent
+				- Some fixture call
+		'''.parse.macroCollection.macros.head.contexts.head as ComponentTestStepContext
+		val reference = TclPackage.eINSTANCE.componentTestStepContext_Component
+
+		// when
+		val actualScope = context.getScope(reference)
+
+		// then
+		assertThat(actualScope).isNotNull
+		assertThat(actualScope.allElements).size.isEqualTo(1)
+		assertThat(actualScope.allElements.get(0).name.toString).isEqualTo("TestName")
+		assertThat(actualScope.allElements.get(0).EClass).isEqualTo(TclPackage.eINSTANCE.macroCollection)
+		assertThat(actualScope.allElements.get(0).EObjectOrProxy).isNotNull
+	}
+}
+
+class TestEditorTestApplication extends XtextApplication<TestEditorConfiguration> {
```

Comment:
looks like too much code duplication with actual implementation, should find a way around that

Reply:
True. The actual Dropwizard application class is already pretty ugly, but I think to properly revise it, we may also have to modify the parent class(es) and re-think their respective responsibilities and the overall inheritance decomposition. Such a re-design would potentially span over several repos...

The corresponding _test_ application class I use here was created by copy-pasting the original and then hard-coding some Mockito magic into it. I suggest that we refactor the _actual_ class, and in the course of doing so, also take testability into account, so it can be parameterized with mocks for testing.